Get lastID for jobs each time we call upgradeDatabase

iwiznia opened the pull request:
@tylerkaraszewski please review.

Fixes 
https://github.com/Expensify/Expensify/issues/77591
https://github.com/Expensify/Expensify/issues/77590
https://github.com/Expensify/Expensify/issues/77592

This happened when master flipped without being restarted.

Is there an easy way to test this? Like maybe changing the priority of a node to become master?

tylerkaraszewski commented:
> Is there an easy way to test this? Like maybe changing the priority of a node to become master?

Do it on the highest priority slave. Bring master down, the slave becomes master. Bring master back up, it will go back to slaving.

iwiznia commented:
ok, updated... In the end, the existing test was already testing this scenario, but it was hitting an edge case in which the initialized ID was 0. Updated the test, it passes now but does not pass without my change.
